cmd/shortener/main.go

```diff
@@ -1,3 +1,51 @@
 package main
 
-func main() {}
+import (
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+	"net/http"
+	"shortener/internal/app/config"
+	createRoute "shortener/internal/app/handlers/create"
+	"shortener/internal/app/handlers/search"
+	"shortener/internal/app/storage"
+	"shortener/internal/app/utils"
+	"time"
+)
+
+func main() {
+	if err := run(); err != nil {
+		panic(err)
+	}
+}
+
+func run() error {
+	// Configuration
+	c := config.NewConfig()
+	r := chi.NewRouter()
+
+	// Middleware
+	r.Use(middleware.Logger)
+	r.Use(middleware.RealIP)
+	r.Use(middleware.Recoverer)
+	r.Use(middleware.RequestID)
+	r.Use(middleware.Timeout(60 * time.Second))
+
+	// Services
+	inMemoryStorage := storage.NewInMemoryStorage()
+
+	// Handlers
+	createHandler := createRoute.New(
+		&utils.UUIDGenerator{},
+		inMemoryStorage,
+		c,
+	)
+	searchHandler := search.New(inMemoryStorage)
+
+	// Routes
+	r.Route("/", func(r chi.Router) {
+		r.Post("/", createHandler.Handle)
+		r.Get("/{id}", searchHandler.Handle)
+	})
+
+	return http.ListenAndServe(c.ServerAddr(), r)
```

Comment:
```suggestion
	log.Fatal(http.ListenAndServe(c.ServerAddr(), r))
```
Так принято - в случае ошибки выводится сама ошибка и программа падает с os.Exit(1). Если ошибки нет - log.Fatal не срабатывает.

Reply:
Исправлено ✅